File: MultiDebugNET/MultiDebug.Server/SrvMessage.cs
namespace MultiDebug.Server
{
    public class SrvMessage
    {
        public MsgType Type { get; }
        public string Content { get; }

        public SrvMessage(MsgType type, string content)
        {
            Type = type;
            Content = content;
        }

        public override string ToString() => $"{Type}:{Content}";

    }
}

File: MultiDebugAndroid/app/src/main/java/com/dmitriykotik/multidebug/client/MultiDebug/Client/SrvMessage.kt
package com.dmitriykotik.multidebug.client.MultiDebug.Client

data class SrvMessage(val type: MsgType, val content: String) {
    companion object {
        fun parse(line: String): SrvMessage? {
            val idx = line.indexOf(':')
            if (idx < 0) return null

            val typeStr = line.substring(0, idx)
            val content = line.substring(idx + 1)

            return try {
                val type = MsgType.valueOf(typeStr)
                SrvMessage(type, content)
            } catch (_: IllegalArgumentException) {
                null
            }
        }
    }
}